### PR TITLE
Add NodeJS and Python providers

### LINKED ISSUE #202: Add other supported generic providers
- [ ] nodejs
- [ ] python

### PR DESCRIPTION
**Pull request by eemcmullan**
Closes #202

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -896,7 +994,9 @@ func (a *analyzeCommand) RunAnalysis(ctx context.Context, xmlOutputDir string, v
 		args = append(args, "--jaeger-endpoint")
 		args = append(args, a.jaegerEndpoint)
 	}
-	if !a.analyzeKnownLibraries {
+
+	// python and node providers do not yet support dep analysis
+	if !a.analyzeKnownLibraries && (providers[0] != pythonProvider && providers[0] != nodeJSProvider) {
```
Comment (shawn-hurley) [inline cmd/analyze.go:999]: We eventually need to not hard code this but be able to tell from the provider caps IMO

Comment (shawn-hurley) [reply cmd/analyze.go:999]: there is nothing to do right now, but something to consider in the future that I didn't want to lose track off

Comment (eemcmullan) [reply cmd/analyze.go:999]: @shawn-hurley https://github.com/konveyor/analyzer-lsp/issues/597

**Comment by shawn-hurley** [comment]
I am wondering if this can be used for the .m2 cache for the Java providers as well.

**Comment by shawn-hurley** [comment]
Overall it looks good, the only big concern is that the python/nodejs servers also still need to have the ability to pull the deps down while initing but that should be solved in those providers.

I wonder if we call it dependency-cache-folder or something similar, the idea being that the underlying thing may add to this?

**Review by shawn-hurley — APPROVED** [review]
LGTM besides minor naming concern

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
nice work! small requests

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**